posthog/api/test/batch_exports/test_delete.py

```diff
@@ -59,6 +64,105 @@ def test_delete_batch_export(client: HttpClient):
         describe_schedule(temporal, batch_export_id)
 
 
+@async_to_sync
+async def wait_for_workflow_executions(
+    temporal: temporalio.client.Client, query: str, timeout: int = 30, sleep: int = 1
+):
+    """Wait for Workflow Executions matching query."""
+    workflows = [workflow async for workflow in temporal.list_workflows(query=query)]
+
+    total = 0
+    while not workflows:
+        total += sleep
+
+        if total > timeout:
+            raise TimeoutError(f"No backfill Workflow Executions after {timeout} seconds")
+
+        await asyncio.sleep(sleep)
+        workflows = [workflow async for workflow in temporal.list_workflows(query=query)]
+
+    return workflows
+
+
+@async_to_sync
+async def wait_for_workflow_in_status(
+    temporal: temporalio.client.Client,
+    workflow_id: str,
+    status: temporalio.client.WorkflowExecutionStatus,
+    sleep: int = 1,
+    timeout: int = 30,
+):
+    """Wait for a Workflow to be in a given status."""
+    handle = temporal.get_workflow_handle(workflow_id)
+    workflow = await handle.describe()
```

Comment:
This all seems logically correct, I'm just curious if I'm reading right that querying the state of workflows counts as an "Action" and is therefore billed at the same level as actually doing workflow steps? Seems really expensive to poll these.

https://docs.temporal.io/workflows#query
https://docs.temporal.io/cloud/pricing#action

Anyway... not really for this PR so much as a higher level question to inform myself.

Reply:
Queries do seem to count as actions for the purposes of billing, but I do not know if `describe` results in a query 🤔

Reply:
The reason why I think it would _not_ count is that queries usually require an implementation on our side to process them, granted that implementation could be done by default by temporal...

Reply:
Coming back to this PR though, we do `describe` and other potentially query-resulting operations in other places, so this could be a bigger scope of reviewing our usage of `describe`/other queries...

Reply:
I've raised a ticket with Temporal support asking for clarification. I've CC'd you in the ticket @bretthoerner. Thank you for raising the point! We can continue the discussion once we hear back from Temporal.